src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryTable.cs

```diff
@@ -377,6 +377,76 @@ private static async Task<(DValue<RecordValue> row, FormulaValue sortValue)> App
             return (row, sortValue);
         }
 
+        private static async Task<(DValue<RecordValue> row, FormulaValue sortValue)> ApplyDistinctLambda(EvalVisitor runner, EvalVisitorContext context, DValue<RecordValue> row, LambdaFormulaValue lambda)
+        {
+            if (!row.IsValue)
+            {
+                return (row, row.ToFormulaValue());
+            }
+
+            var childContext = context.SymbolContext.WithScopeValues(row.Value);
+            var distinctValue = await lambda.EvalInRowScopeAsync(context.NewScope(childContext));
+
+            return (row, distinctValue);
+        }
```

Comment:
Can the ApplySortLambda() be resused? maybe rename that one?

Reply:
Yeah. I think I forked it thinking they would have to be different and then they weren't. Good catch.